### PR TITLE
Add negative tests for json types with table types

### LINKED ISSUE #32397: Getting NPE when trying to add table for a json type
**Description:**
I tried to add a table to a json like in below.

```ballerina
    json j = {
        n: table key(id) [
            {id: 1, firstName: "John", lastName: "Smith", salary: 100},
            {id: 2, firstName: "Fred", lastName: "Bloggs", salary: 200}
        ]
    };
```

Then I found json is not accepting tables according to the spec [1].

However this is giving the following null pointer exception at the compilation and we might need to improve this case.

```
[2021-08-26 19:20:07,713] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.visit(CodeAnalyzer.java:3019)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangRecordLiteral.accept(BLangRecordLiteral.java:72)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.analyzeExpr(CodeAnalyzer.java:3983)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.analyzeExprs(CodeAnalyzer.java:4042)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.visit(CodeAnalyzer.java:3000)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangTableConstructorExpr.accept(BLangTableConstructorExpr.java:67)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.analyzeExpr(CodeAnalyzer.java:3983)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.visit(CodeAnalyzer.java:3008)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangRecordLiteral.accept(BLangRecordLiteral.java:72)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.analyzeExpr(CodeAnalyzer.java:3983)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.visit(CodeAnalyzer.java:2521)
        at org.wso2.ballerinalang.compiler.tree.BLangSimpleVariable.accept(BLangSimpleVariable.java:53)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.analyzeNode(CodeAnalyzer.java:394)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.visit(CodeAnalyzer.java:2616)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangSimpleVariableDef.accept(BLangSimpleVariableDef.java:46)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.analyzeNode(CodeAnalyzer.java:394)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.visit(CodeAnalyzer.java:574)
        at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:57)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.analyzeNode(CodeAnalyzer.java:394)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.visitFunction(CodeAnalyzer.java:542)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.visit(CodeAnalyzer.java:495)
        at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:73)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.analyzeNode(CodeAnalyzer.java:394)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.analyzeTopLevelNodes(CodeAnalyzer.java:382)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.visit(CodeAnalyzer.java:374)
        at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:165)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CodeAnalyzer.analyze(CodeAnalyzer.java:363)
        at io.ballerina.projects.internal.CompilerPhaseRunner.codeAnalyze(CompilerPhaseRunner.java:212)
        at io.ballerina.projects.internal.CompilerPhaseRunner.performTypeCheckPhases(CompilerPhaseRunner.java:117)
        at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:383)
        at io.ballerina.projects.ModuleCompilationState$1.compile(ModuleCompilationState.java:45)
        at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:329)
        at io.ballerina.projects.PackageCompilation.compileModulesInternal(PackageCompilation.java:174)
        at io.ballerina.projects.PackageCompilation.compileModules(PackageCompilation.java:166)
        at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:95)
        at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:206)
        at io.ballerina.projects.Package.getCompilation(Package.java:135)
        at io.ballerina.cli.task.CompileTask.execute(CompileTask.java:64)
        at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
        at io.ballerina.cli.cmd.RunCommand.execute(RunCommand.java:163)
        at java.base/java.util.Optional.ifPresent(Optional.java:183)
        at io.ballerina.cli.launcher.Main.main(Main.java:58)
```

[1] https://ballerina.io/spec/lang/2021R1/#json-type-descriptor

**Steps to reproduce:**
Add the above json declaration and run `bal build`

**Affected Versions:**
```
Ballerina Swan Lake Beta 2
Language specification 2021R1
Update Tool 1.3.5
```

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Fixes #32397

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by KavinduZoysa — APPROVED** [review]

**Review by lasinicl — DISMISSED** [review]

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/json/TableToJSONNegaiveTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package org.ballerinalang.test.types.json;
```
Comment (lasinicl) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/json/TableToJSONNegaiveTest.java:1]: Add license header

Comment (SasinduDilshara) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/json/TableToJSONNegaiveTest.java:17]: Updated

**Comment by SasinduDilshara** [comment]
> @SasinduDilshara, let's fix the build failure.

fix ubuntu and windows failure

File: tests/jballerina-unit-test/src/test/resources/test-src/types/jsontype/table_to_json_negative.bal [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (lasinicl) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/jsontype/table_to_json_negative.bal:1]:
```suggestion
// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: tests/jballerina-unit-test/src/test/resources/test-src/types/jsontype/table_to_json_negative.bal [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        ]
+    };
+
+    json j2 = <json> {
+        n: table key(id) [
+            {id: 1, firstName: "John", lastName: "Smith", salary: 100},
+            {id: 2, firstName: "Fred", lastName: "Bloggs", salary: 200}
+        ]
+    };
+}
```
Comment (KavinduZoysa) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/jsontype/table_to_json_negative.bal:45]: Shall we add a few tests without key and empty table?

Comment (SasinduDilshara) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/jsontype/table_to_json_negative.bal:45]: I will add

Comment (SasinduDilshara) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/jsontype/table_to_json_negative.bal:75]: Added

**Review by lasinicl — APPROVED** [review]

**Review by KavinduZoysa — APPROVED** [review]
